### ISSUE TITLE
TPOT Overfitting

### ISSUE THREAD

**Issue by jmrichardson** [EpistasisLab/tpot#853]
Hello, I am new to TPOT and running into an overfitting issue.  The pipeline selected "ExtraTreeClassifier" got an almost perfect CV score:

```
print(tpot.score(X, y))
0.9999142532530172
```

I know this is impossible so I am wondering the following:

1: How can I look at the accuracy for both training and cross validation?  I have looked at the tpotClassifier object and I see the internal_cv_score for each individual but can't find the training score.  I would like to be able to look at the differences to see if there is an indication of overfitting
2: I was under the assumption that TPOT automatically chooses the best model that avoids overfitting?

I am sure I am doing something wrong and could use a pointer.  

## Context of the issue

Here is my TPOTClassifier config.  I am testing so only 1 generation.  If I remove the ExtraTreeClassifier, it selects a model with score of .63 which is close to what I get just running an sklearn RF model:

```
gkf = GroupKFold(n_splits=4).split(X=X, y=y, groups=groups)

tpot_config = {
    'sklearn.ensemble.RandomForestClassifier': {
        'n_estimators': [10, 20],
        'max_features': ['auto', 'sqrt', 'log2'],
        'min_samples_leaf': [5,10],
        'max_depth': [10, None],
    },
    'sklearn.tree.ExtraTreeClassifier': {
    }
}

tpot = TPOTClassifier(generations=1, population_size=30, verbosity=3,
                     cv=gkf,
                     periodic_checkpoint_folder='tmp',
                     scoring='accuracy',
                     early_stop=5,
                     random_state=1,
                     memory='tmp',
                     warm_start=True,
                     config_dict=tpot_config
                     )

tpot.fit(X, y)
print(tpot.score(X, y))
```

**Comment by jmrichardson** [comment]
I think the issue was the way I was splitting for GroupKFold.  Seems to be working fine now.  However, I would still like to know if there is a way to see the training accuracy in addition to the cv score?

```
groups = df['id']
tpot_config = {
    'sklearn.ensemble.RandomForestClassifier': {
        'n_estimators': [10, 20],
        'max_features': ['auto', 'sqrt', 'log2'],
        'min_samples_leaf': [5,10],
        'max_depth': [10, None],
    },
    'sklearn.tree.ExtraTreeClassifier': {
    }
}

# Train model
tpot = TPOTClassifier(generations=2, population_size=5, verbosity=3,
                     cv=GroupKFold(n_splits=3),
                     periodic_checkpoint_folder='tmp',
                     scoring='accuracy',
                     early_stop=2,
                     random_state=1,
                     memory='tmp',
                     warm_start=True,
                     config_dict=tpot_config
                     )

# tpot.fit(X, y)
tpot.fit(X, y, groups=groups)
```

**Comment by weixuanfu** [comment]
The current version of TPOT doesn't report/store training accuracy during optimization. Maybe it is a useful enhancement function.
